### PR TITLE
chore: make github app optional

### PR DESCRIPTION
**Pull request by Eun**
Hey, I am trying to self host vrite and wondered if selfhosters really need all the dependencies the full app on vrite.io needs.

So this (if you accept this pr) would be the first pr of a series of pull requests to disable all the fancy stuff a self hoster probably not need or want.

### CODE REVIEW

**Comment by areknawo** [comment]
Hey, thanks for the PR.

That's the first external one, so I'm definitely very grateful ❤️

That said, I've been working on a `docker-compose.yml` file and a proper guide for self-hosting this week, so good timing 🙂 I hope to have it released next week around Tuesday (definitely `docker-compose.yml` and hopefully the docs) and this will absolutely include optional env vars.

Here's the plan:
- Allow to choose between SendGrid or custom SMTP configuration;
- Allow to omit GitHub OAuth2 credentials;
- Allow to omit GitHub App credentials;
- Allow to omit OpenAI (and potentially Weaviate);
- Based on the available env vars, enable/disable different endpoints;
- Make the "enabled features" endpoint to let the dashboard know which features are enabled/disabled, and configure the UI accordingly;
- MongoDB, Redis, S3-compatible bucket are still required;

The PR makes the properties optional and only omits the GitHub App connection plugin, while technically, omitting the GitHub App (at least for now), disables the entire Git sync feature. So, while it shouldn't be breaking, there can still be unexpected errors when trying to use the Git sync panel from the UI.

TL;DR; You'll likely want to wait till next week. But also, let me know if you have any suggestions to make the self-hosting experience better, considering the above plan.
